### PR TITLE
Added some notes related to spring based projects on Neo platform.

### PR DESCRIPTION
**Pull request by TanviiG**
## What Has Changed?

The PR adds notes to highlight that we only support servlet based applications on Neo environment and spring applications are not supported out-of-the-box.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on it's own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [ ] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)

### CODE REVIEW

File: docs/java/getting-started.mdx [hunk 1]
```diff
@@ -130,6 +130,10 @@ skipUsageAnalytics: false
 
 **Congratulations! You've just set up a brand new application with the SAP Cloud SDK for Java.**
 
+:::note
```
Comment (MatKuhr) [inline docs/java/getting-started.mdx:133]: I would actually put this higher up next to the code snippet and mention that the snippets are for CF & that for neo only the servlet based approach is supported.

Comment (MatKuhr) [reply docs/java/getting-started.mdx:133]: otherwise lgtm

Comment (TanviiG) [reply docs/java/getting-started.mdx:130]: done

Comment (MatKuhr) [reply docs/java/getting-started.mdx:130]: No I meant really next to the code snippet, not after the log output. If people really want to choose NEO mentioning it here is too late, at this point in the article the archetype generation is already done.

**Comment by artemkovalyov** [comment]
@TanviiG , can you address the latest notes from @MattKuhr?

File: docs/java/getting-started.mdx (hunk 1 above)
Comment (TanviiG) [reply docs/java/getting-started.mdx]: Done.. please take a look if it is ok.

**Comment by TanviiG** [comment]
> @TanviiG , can you address the latest notes from @MattKuhr? Then it's good to be merged.

Yes I have already addressed it (just did it half an hour ago).

**Review by artemkovalyov — APPROVED** [review]
LGTM

**Pull request merged**